lib/ansible/plugins/httpapi/qradar.py

```diff
@@ -0,0 +1,67 @@
+# (c) 2019 Red Hat Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+author: Ansible Security Automation Team
+httpapi : qradar
+short_description: HttpApi Plugin for IBM QRadar appliances
+description:
+  - This HttpApi plugin provides methods to connect to IBM QRadar
+    appliances over a HTTP(S)-based api.
+version_added: "2.8"
+"""
+
+import json
+
+from ansible.module_utils.basic import to_text
+from ansible.errors import AnsibleConnectionFailure
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.plugins.httpapi import HttpApiBase
+from ansible.module_utils.connection import ConnectionError
+
+# Content Type and QRadar REST API Version
+BASE_HEADERS = {
+    'Content-Type': 'application/json;charset=UTF-8',
+    'Version': '9.1',
+}
+
+
+class HttpApi(HttpApiBase):
+    def logout(self):
+        response, dummy = self.send_request('POST', '/api/auth/logout')
+
+    def send_request(self, request_method, path, payload=None):
+        data = json.dumps(payload) if payload else '{}'
+
+        try:
+            self._display_request(request_method)
+            response, response_data = self.connection.send(path, payload, method=request_method, headers=BASE_HEADERS, force_basic_auth=True)
+            value = self._get_response_value(response_data)
+
+            return response.getcode(), self._response_to_json(value)
+        except AnsibleConnectionFailure as e:
+            if to_text('401') in to_text(e):
```

Comment:
You should be able to deal with all of this more sensibly if you override `handle_httperror()`... see https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/httpapi/__init__.py#L54